wallet/wallet_test.go

```diff
@@ -205,3 +206,89 @@ func TestLabelTransaction(t *testing.T) {
 		})
 	}
 }
+
+// TestGetTransaction tests if we can fetch a mined, an existing
+// and a non-existing transaction from the wallet like we expect.
+func TestGetTransaction(t *testing.T) {
+
+	tests := []struct {
+		name string
+
+		// Whether the transaction should be known to the wallet.
+		txKnown bool
+
+		// Whether the transaction is mined.
+		minedTx bool
+
+		// Block meta. Used to emulate mined transactions.
+		blockMeta *wtxmgr.BlockMeta
+
+		// The error we expect to be returned.
+		expectedErr error
+	}{
+		{
+			name:        "existing transaction, not mined",
+			txKnown:     true,
+			blockMeta:   nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "non-existing transaction",
+			txKnown:     false,
+			blockMeta:   nil,
+			expectedErr: ErrNoTx,
+		},
+		{
+			name:    "existing mined transaction",
+			txKnown: true,
+			blockMeta: &wtxmgr.BlockMeta{
+				Block: wtxmgr.Block{
+					Height: 100,
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			w, cleanup := testWallet(t)
+			defer cleanup()
+
+			// If the transaction should be known to the store, we
+			// write txdetail to disk.
+			if test.txKnown {
+				rec, err := wtxmgr.NewTxRecord(
+					TstSerializedTx, time.Now(),
+				)
+				require.NoError(t, err)
+
+				// Write the tx record to disk.
+				err = walletdb.Update(w.db,
+					func(tx walletdb.ReadWriteTx) error {
+
+						ns := tx.ReadWriteBucket(
+							wtxmgrNamespaceKey,
+						)
+
+						return w.TxStore.InsertTx(ns, rec, test.blockMeta)
+
+					})
+				require.NoError(t, err, "could not insert tx")
+			}
+
+			tx, err := w.GetTransaction(*TstTxHash)
+			if err != test.expectedErr {
+				t.Fatalf("expected: %v, got: %v", test.expectedErr, err)
+			}
+			if test.txKnown {
+				require.True(t, tx.Summary.Hash.IsEqual(TstTxHash))
+			}
+			// Test if the transaction is mined.
+			if test.blockMeta != nil {
```

Comment:
This is testing that when it's mined, the `Height` field is set, but it's not testing when it's not mined, this field is not set.